### PR TITLE
Prometheus: Add tooltip to explain possibility to use patterns in text and title fields in annotations

### LINKED ISSUE #29451: Annotations prometheus query Title and Text not working without double curly brackets (Missing documentation)
**What would you like to be added**:

I added an annotations query to my dashboard using prometheus as datasource.
In `Field formats` the `Title` and `Text` fields did not use the provided prometheus label correctly, `Tags` were working fine.
The entered label-name was used literally in the ui (see below).

**While writing this issue, I tried the `{{label}}` syntax that is used in panel legends and it works.** ☑
This however is not documented:
- https://grafana.com/docs/grafana/latest/datasources/prometheus/#annotations
- https://grafana.com/docs/grafana/latest/dashboards/annotations/#querying-other-data-sources

and unexpected, since escaping is not needed for the `Tags` property, but comprehensible since the user may enter other text besides the label.

Has this always been like this? If yes then
- documentation should cover this (I could do a PR for this, just @ me),
- and also the UI textbox help label should use the `{{label}}` syntax as example, not sure if I can find the correct source for this.

**Why is this needed**:

Undocumented behavior / hidden features may frustrate users.

ok-config

config-suggestion

**How to reproduce it (as minimally and precisely as possible)**:

( I started with the bug report template..)

- Go to: https://play.grafana.org/d/000000029 / Settings / Annotations / **Add Annotation Query**
- Use Data source `demo.robustperception.io` or any prometheus that has active alerts
- Use Expression: `ALERTS` , Title: `alertname` , Text: `job` , Tags: `alertstate` , hit `Add`
- Go back, wait for dashboard to refresh, mouse hover on the displayed alert:
  - Tooltip displays "alertname" and "job" ❌ , but should display (for this case) `ExampleAlertAlwaysFiring` and `pushgateway`
  - The `alertstate` Tag is displayed correctly ✔
- Change any panel query to `ALERTS` to view currently active alerts of this data source

wrong alert display

wrong-config

active alerts

**Environment**:
https://play.grafana.org
- Grafana version: `v7.3.3 (2489dc4d3a)`
- Data source type & version: prometheus
- User OS & Browser: Firefox

### PR DESCRIPTION
**Pull request by ivanahuckova**
Fixes: https://github.com/grafana/grafana/issues/29451

Related documentation fix: https://github.com/grafana/website/pull/3353

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/partials/annotations.editor.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
 </div>
 
 <div class="gf-form-group">
-	<h5 class="section-heading">Field formats</h5>
+	<h5 class="section-heading">Field formats<tip>For title and text fields you can choose to use the name or pattern. For example {{instance}} will be replaced with label value for the label instance.</tip></h5>
```
Comment (achatterjee-grafana) [inline public/app/plugins/datasource/prometheus/partials/annotations.editor.html:13]:
```suggestion
	<h5 class="section-heading">Field formats<tip>For title and text fields, use either the name or a pattern. For example, {{instance}} is replaced with label value for the label instance.</tip></h5>
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added copy-edits.

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

**Comment by oddlittlebird** [comment]
@ivanahuckova, is this a new feature that should be added to What's new in 7.4?

**Comment by ivanahuckova** [comment]
@oddlittlebird no, it is a very old feature, we just weren't clear about how you can use pattern in it (the same pattern as we use as legendFormat. So I added a tooltip and section to documentation that clarifies it.

**Comment by oddlittlebird** [comment]
Works for me!

**Pull request merged**